tests/queue.spec.ts

```diff
@@ -150,6 +151,41 @@ describe('ARRAY LIKE USAGE', () => {
 	});
 });
 
+describe('Iteration', () => {
+	it('should work with empty queue', () => {
+		const iter = new QueueIterator(queue);
+		expect(() => {
+			const res = iter.next();
+			expect(res.value).toBe(null);
+			expect(res.done).toBe(true);
+		}).not.toThrow();
+	});
+
+	it('should work with one element', () => {
+		queue.push('string');
+		const iter = new QueueIterator(queue);
+		let res = iter.next();
+		expect(res.value).toBe('string');
+		expect(res.done).toBe(false);
+		res = iter.next();
+		expect(res.value).toBe(null);
+		expect(res.done).toBe(true);
+	});
+
```

Comment:
We've got the `0` and `1` element cases covered once the above feedback is done! Now, we need the arbitrary `n` case, where `n` is a small-ish but reasonably sized (relative to this collection) set of items. Let's pick 11. It's an odd number.

- [ ] Implement the same set of tests used above for an empty queue, 1 element queue, but with 11 elements.